inventory_management_system_api/routers/v1/catalogue_item.py

```diff
@@ -155,6 +156,10 @@ def partial_update_catalogue_item(
         message = "Adding a catalogue item to a non-leaf catalogue category is not allowed"
         logger.exception(message)
         raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=message) from exc
+    except ChildrenElementsExistError as exc:
+        message = "Catalogue item has child elements and cannot be updated"
+        logger.exception(message)
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=message) from exc
```

Comment:
Worth clarifying with Viktor, do we need to prevent this completely? - to my mind the only thing that could cause issues are changing the names/adding or removing properties, the values are confirmed when the items are added so they are just defaults in the catalogue item, and the others are not stored anywhere but here. Potentially a future issue but we may be able to open it up again similar to https://github.com/ral-facilities/inventory-management-system-api/pull/98.

Reply:
Yeah I don't think it would be too bad, its really a case of preventing it when the items are modified in anyway but it has children, which should end up a similar size to #98. I get the feeling that it would be something they want given we already had to add it in case a catalogue category needed renaming or something. I would suggest we should just create an issue for it and think about it again later.

Reply:
After thinking about this a bit more earlier this morning, it doesn't actually seem too bad. We should probably address this and I don't mind if this is done in this PR given the size. Equally happy to be addressed separately.

Reply:
I'll create an issue and address it in a separate PR, to clarify we would want to only check for children items if they are trying to update catalogue item properties?

Reply:
Issue created https://github.com/ral-facilities/inventory-management-system-api/issues/153